include/boost/gil/image_processing/harris.hpp

```diff
@@ -0,0 +1,74 @@
+#ifndef BOOST_GIL_HARRIS_HPP
+#define BOOST_GIL_HARRIS_HPP
+
+#include <boost/gil/image_view.hpp>
+#include <boost/gil/typedefs.hpp>
+
+namespace boost { namespace gil {
+/// \defgroup CornerDetectionAlgorithms
+/// \brief Algorithms that are used to find corners in an image
+///
+/// These algorithms are used to find spots from which
+/// sliding the window will produce large intensity change
+
+
+/// \brief function to record Harris responses
+/// \ingroup CornerDetectionAlgorithms
+///
+/// This algorithm computes Harris responses
+/// for structure tensor represented by m11, m12_21, m22 views.
+/// Note that m12_21 represents both entries (1, 2) and (2, 1).
+/// Window length represents size of a window which is slided around
+/// to compute sum of corresponding entries. k is a discrimination
+/// constant against edges (usually in range 0.04 to 0.06).
+/// harris_response is an out parameter that will contain the Harris responses.
+void compute_harris_responses(
+    gil::gray32f_view_t m11,
+    gil::gray32f_view_t m12_21,
+    gil::gray32f_view_t m22,
+    gil::gray32f_view_t weights,
+    float k,
+    gil::gray32f_view_t harris_response)
+{
+    if (m11.dimensions() != m12_21.dimensions() || m12_21.dimensions() != m22.dimensions()) {
+        throw std::invalid_argument("m prefixed arguments must represent"
+            " tensor from the same image");
+    }
+
+    auto window_length = weights.dimensions().x;
+    auto width = m11.width();
+    auto height = m11.height();
```

Comment:
Should read `auto const width =...`, for all immutable variables. This allows a reader of this code to reason with one or two questions less about the code.